homeassistant/components/wsdot/sensor.py

```diff
@@ -63,10 +64,14 @@ def setup_platform(
     sensors = []
     for travel_time in config[CONF_TRAVEL_TIMES]:
         name = travel_time.get(CONF_NAME) or travel_time.get(CONF_ID)
-        sensors.append(
-            WashingtonStateTravelTimeSensor(
-                name, config.get(CONF_API_KEY), travel_time.get(CONF_ID)
+        api_key = config.get(CONF_API_KEY)
+        if not api_key or not isinstance(api_key, str):
+            raise ConfigEntryError(
+                f"wsdot requires an {CONF_API_KEY}. "
+                "See https://www.home-assistant.io/integrations/wsdot#setup"
             )
```

Comment:
API key is always required so I'm not sure what this logic is for

Reply:
It is required, but mypy thought that `config.get(CONF_API_KEY)` was `Any`, so I did this rather than `assert` to type narrow.

Reply:
Then we should just do `config[CONF_API_KEY]`